### PR TITLE
adds `added_on` field to files

### LINKED ISSUE #1228: Add `added_at` and `claim_updated_at` columns to the file_list output
`added_at` would represent the time the file was added to the database. 

`claim_updated_at` will be added by fixing #2086, which will add the associated Claim object to each file, and the Claim object has a `timestamp` field.

After adding these fields to the database they'd be added as filters to the `file_x` api calls as well.

steps:
- [ ] write migration to add `added_at` column to the sqlite db
- [ ] add code to set `added_at` field for new files
- [ ] fix #2086 
- [ ] add API params for filtering and sorting by these fields

### PR DESCRIPTION
**Pull request by osilkin98**
fixes #1228

### CODE REVIEW

**Comment by lyoshenka** [comment]
@osilkin98 whats the next step on this?

**Comment by osilkin98** [comment]
Testing migration right now, should be ready after.

**Comment by osilkin98** [comment]
I haven't added `claim_updated_at` since Jack said it can be derived from the block height, so no need to store it. 

Aside from that, this PR is ready to be merged.

File: lbry/lbry/extras/daemon/storage.py [hunk 1]
```diff
@@ -247,7 +251,8 @@ class SQLiteStorage(SQLiteMixin):
                 blob_data_rate real not null,
                 status text not null,
                 saved_file integer not null,
-                content_fee text
+                content_fee text,
+                added_at integer
```
[the remark this answers is not in the document]
Comment (osilkin98) [reply lbry/lbry/extras/daemon/storage.py]: gotcha

File: lbry/lbry/extras/daemon/migrator/migrate11to12.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import os
+
+
+def do_migration(conf):
+    db_path = os.path.join(conf.data_dir, 'lbrynet.sqlite')
+    connection = sqlite3.connect(db_path)
+    connection.row_factory = sqlite3.Row
+    cursor = connection.cursor()
+    cursor.executescript('alter table file add column added_at integer')
+    cursor.execute("UPDATE file SET added_at = strftime('%s', 'now')")
```
[the remark this answers is not in the document]
Comment (osilkin98) [reply lbry/lbry/extras/daemon/migrator/migrate11to12.py]: Done.

**Comment by jackrobison** [comment]
@osilkin98 I think this is good to go - can you rebase/squash this down? Then we'll merge

@tzarebczan can you sanity test this with a large database?

**Comment by tzarebczan** [comment]
Get this when running this branch: ```future: <Task finished coro=<Component._setup() done, defined at c:\users\htpc\lbry\lbry\lbry\extras\daemon\Component.py:53> exception=AttributeError("'NoneType' object has no attribute 'sync_missing_blobs'")>
Traceback (most recent call last):
  File "c:\users\htpc\lbry\lbry\lbry\extras\daemon\Component.py", line 63, in _setup
    raise err
  File "c:\users\htpc\lbry\lbry\lbry\extras\daemon\Component.py", line 55, in _setup
    result = await self.start()
  File "c:\users\htpc\lbry\lbry\lbry\extras\daemon\Components.py", line 292, in start
    return await self.blob_manager.setup()
  File "c:\users\htpc\lbry\lbry\lbry\blob\blob_manager.py", line 86, in setup
    to_add = await self.storage.sync_missing_blobs(in_blobfiles_dir)
AttributeError: 'NoneType' object has no attribute 'sync_missing_blobs'
2019-10-02 18:04:41,411 INFO     lbry.extras.daemon.Components:448: start blob server
2019-10-02 18:04:41,420 INFO     lbry.extras.daemon.Components:386: Started blob announcer
2019-10-02 18:04:41,421 INFO     lbry.extras.daemon.Components:423: Starting the file manager
2019-10-02 18:04:41,423 ERROR    lbry.extras.daemon.Component:62: Error setting up stream_manager
Traceback (most recent call last):
  File "c:\users\htpc\lbry\lbry\lbry\extras\daemon\Component.py", line 55, in _setup
    result = await self.start()
  File "c:\users\htpc\lbry\lbry\lbry\extras\daemon\Components.py", line 428, in start
    await self.stream_manager.start()
  File "c:\users\htpc\lbry\lbry\lbry\stream\stream_manager.py", line 200, in start
    await self.load_and_resume_streams_from_database()
  File "c:\users\htpc\lbry\lbry\lbry\stream\stream_manager.py", line 138, in load_and_resume_streams_from_database
    await self.storage.update_manually_removed_files_since_last_run()
AttributeError: 'NoneType' object has no attribute 'update_manually_removed_files_since_last_run'
2019-10-02 18:04:41,436 INFO     lbry.blob_exchange.server:168: Blob server listening on TCP 0.0.0.0:3336
Task exception was never retrieved
future: <Task finished coro=<Component._setup() done, defined at c:\users\htpc\lbry\lbry\lbry\extras\daemon\Component.py:53> exception=AttributeError("'NoneType' object has no attribute 'update_manually_removed_files_since_last_run'")>
Traceback (most recent call last):
  File "c:\users\htpc\lbry\lbry\lbry\extras\daemon\Component.py", line 63, in _setup
    raise err
  File "c:\users\htpc\lbry\lbry\lbry\extras\daemon\Component.py", line 55, in _setup
    result = await self.start()
  File "c:\users\htpc\lbry\lbry\lbry\extras\daemon\Components.py", line 428, in start
    await self.stream_manager.start()
  File "c:\users\htpc\lbry\lbry\lbry\stream\stream_manager.py", line 200, in start
    await self.load_and_resume_streams_from_database()
  File "c:\users\htpc\lbry\lbry\lbry\stream\stream_manager.py", line 138, in load_and_resume_streams_from_database
    await self.storage.update_manually_removed_files_since_last_run()
AttributeError: 'NoneType' object has no attribute 'update_manually_removed_files_since_last_run'
2019-10-02 18:04:54,927 INFO     lbry.dht.node:158: Joined DHT, 27 peers known in 5 buckets
2019-10-02 18:05:32,944 INFO     torba.client.baseledger:136: lbc_mainnet: added BlockHeightEvent(height=643421, change=1) header blocks, final height 643421
2019-10-02 18:05:41,267 INFO     torba.client.baseledger:136: lbc_mainnet: added BlockHeightEvent(height=643422, change=1) header blocks, final height 643422
2019-10-02 18:07:16,553 INFO     torba.client.baseledger:136: lbc_mainnet: added BlockHeightEvent(height=643423, change=1) header blocks, final height 643423
2019-10-02 18:07:46,326 INFO     lbry.extras.daemon.Components:390: Stopped blob announcer
2019-10-02 18:07:46,327 INFO     lbry.blob_exchange.server:178: Stopped blob server
2019-10-02 18:07:46,332 INFO     lbry.dht.node:111: Stopped DHT node
2019-10-02 18:07:46,337 INFO     torba.client.basenetwork:190: Wallet client switching task stopped.
2019-10-02 18:07:46,369 INFO     lbry.extras.daemon.exchange_rate_manager:218: Stopping exchange rate manager
2019-10-02 18:07:46,370 INFO     lbry.extras.daemon.Daemon:465: stopped api components
2019-10-02 18:07:46,372 INFO     lbry.extras.daemon.Daemon:471: stopped api server
2019-10-02 18:07:46,372 INFO     lbry.extras.daemon.Daemon:474: finished shutting down
Task exception was never retrieved
future: <Task finished coro=<BlobAnnouncer._announce() done, defined at c:\users\htpc\lbry\lbry\lbry\dht\blob_announcer.py:31> exception=AttributeError("'NoneType' object has no attribute 'get_blobs_to_announce'")>
Traceback (most recent call last):
  File "c:\users\htpc\lbry\lbry\lbry\dht\blob_announcer.py", line 39, in _announce
    self.announce_queue.extend(await self.storage.get_blobs_to_announce())
AttributeError: 'NoneType' object has no attribute 'get_blobs_to_announce'```

**Comment by tzarebczan** [comment]
This looks good to me now - no longer failing after Oleg fixed up a few things. 

Do we want the column name to be "added_at"  (current), or "created_at"? I'm ok with either.

**Comment by tzarebczan** [comment]
Also tested while killing the migration - was able to run it successfully after. Ctrl-C would not kill it though, had to manually do it.

**Comment by osilkin98** [comment]
`created_at` is a bit ambiguous since it can be interpreted as the time when the stream was created

**Comment by tzarebczan** [comment]
What's the next step here?

**Pull request merged**